test/location.js

```diff
@@ -0,0 +1,185 @@
+            t.ok(body)
+            t.equals(2, body.total, 'total should be two')
+            t.equals(body.resourceType, 'Bundle', 'should return a Bundle')
+            t.equals(body.entry[0].resource.resourceType, 'Location', 'should return a resource of type Location')
+            t.equals(body.entry[0].resource.name, 'Greenwood Clinical Practice', 'should return resource with name Greenwood')
+            t.equals(body.entry[1].resource.resourceType, 'Location', 'should return a resource of type Location')
+            t.equals(body.entry[1].resource.name, 'Yellow wood clinic', 'should return resource with name Greenwood')
+            done()
+          })
+        })
+      })
+    })
+  })
+
+  t.test('should support _summary parameter on read', (t) => {
+    setupTestEnv(t, (db, done) => {
+      // given
+      const location1 = _.cloneDeep(locationTemplate)
+      delete location1.id
+      env.createResource(t, location1, 'Location', (err, ref) => {
+        t.error(err)
+
+        // when
+        request({
+          url: `http://localhost:3447/fhir/Location/${ref.split('/').pop()}?_summary=true`,
+          headers: headers,
+          json: true
+        }, (err, res, body) => {
+          // then
+          t.error(err)
+
+          t.equal(res.statusCode, 200, 'response status code should be 200')
+          t.ok(body)
+          t.equals(body.name, 'Greenwood Clinical Practice', 'should return correct resource')
+
+          t.notOk(body.telecom)
+          t.notOk(body.address)
+          t.notOk(body.position)
+
+          done()
+        })
+      })
+    })
+  })
+
+  t.test('should read hook should return all data when no _summary param is present', (t) => {
+    setupTestEnv(t, (db, done) => {
+      // given
+      const location1 = _.cloneDeep(locationTemplate)
+      delete location1.id
+      env.createResource(t, location1, 'Location', (err, ref) => {
+        t.error(err)
+
+        // when
+        request({
+          url: `http://localhost:3447/fhir/Location/${ref.split('/').pop()}`,
+          headers: headers,
+          json: true
+        }, (err, res, body) => {
+          // then
+          t.error(err)
+
+          t.equal(res.statusCode, 200, 'response status code should be 200')
+          t.ok(body)
+          t.equals(body.name, 'Greenwood Clinical Practice', 'should return correct resource')
+
+          t.ok(body.telecom)
+          t.ok(body.address)
+          t.ok(body.position)
+
+          done()
+        })
+      })
+    })
+  })
+
+  t.test('should return 400 on unsupported parameter', (t) => {
+    setupTestEnv(t, (db, done) => {
+      // when
+      request({
+        url: 'http://localhost:3447/fhir/Location?notsupported=meh',
+        headers: headers,
+        json: true
+      }, (err, res, body) => {
+        // then
+        t.error(err)
+
+        t.equal(res.statusCode, 400, 'response status code should be 400')
+        done()
+      })
+    })
+  })
+})
```

Comment:
I wonder if we should add tests for read and insert requests, at the moment this is only testing search interactions

Reply:
Ignore the above comment, these tests are handled